pylint/lint/parallel.py

```diff
@@ -109,7 +115,7 @@ def _merge_mapreduce_data(linter, all_mapreduce_data):
             checker.reduce_map_data(linter, collated_map_reduce_data[checker.name])
 
 
-def check_parallel(linter, jobs, files, arguments=None):
+def check_parallel(linter, jobs, files: Iterator[FileItem], arguments=None):
```

Comment:
Does it need to be an `Iterator` or is `Iterable` enough?
https://docs.python.org/3/library/collections.abc.html#collections.abc.Iterable